### PR TITLE
✨  Legend Component

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

Creates a Legend component for the File Entity & Donor Entity pages
Updates the Dropdown button to include a controlled menu state which allows parent/children of the button to access menu state
Adds ability to style inner menu
Adds Legend Icon

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/uikit/25

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [x] Modified Existing components and updates stories
  - [ ] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: uikit/DropdownButton/index.tsx [hunk 1]
```diff
@@ -49,43 +49,57 @@ type DropdownButtonItemConfig<ValueType = string> = {
 };
 export type DownloadButtonProps<ValueType> = {
   onItemClick: (item: DropdownButtonItemConfig<ValueType>) => void;
+  onClick?:
+    | ((e: React.SyntheticEvent<HTMLButtonElement, Event>) => any)
+    | ((e: React.SyntheticEvent<HTMLButtonElement, Event>, {}) => any);
```
Comment (hlminh2000) [inline uikit/DropdownButton/index.tsx:54]: I think it's less confusing to just type this as:
```
onClick: ((e: React.SyntheticEvent<HTMLButtonElement, Event>, {toggleMenuOpen: boolean}) => any)
``` 
:)

Comment (demariadaniel) [reply uikit/DropdownButton/index.tsx]: Agreed, updated

File: uikit/Button/index.tsx [hunk 2]
```diff
@@ -79,6 +79,7 @@ const Button = React.forwardRef<
       Loader,
       type,
       showLoaderWithChildren,
+      ...rest
```
Comment (hlminh2000) [inline uikit/Button/index.tsx:82]: probably will need to extend the prop's type to be `React.ComponentProps<typeof StyledButton> & {...}` that way all the `StyledButton`'s props are also exposed through the `Button` :P

lol slow day in the office for me here...

Comment (demariadaniel) [reply uikit/Button/index.tsx:82]: Hey @hlminh2000! @joneubank  & I took a look at this. 

I've tried updating `interface DownloadButtonProps<ValueType>` to `extends Omit<React.ComponentProps<typeof StyledButton>, 'onClick'>` but this creates type issues with other usages of the Download Button.

Jon pointed out that extending StyledButton would protect against future updates, but he validated the types/props should work as is.

**Review by ciaranschutte — APPROVED** [review]

**Pull request merged**